src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala

```diff
@@ -134,25 +147,39 @@ class WorkflowManagerActor(dataAccess: DataAccess, backend: Backend) extends Act
     }
   }
 
-  private def callStdoutStderr(workflowId: WorkflowId, callFqn: String, index: ExecutionIndex): Future[StdoutStderr] = {
+  private def isCollector(entries: Iterable[ExecutionDatabaseKey])(key: ExecutionDatabaseKey) = {
+    key.index.isEmpty &&
+      (entries exists { e =>
+        (e.fqn == key.fqn) && e.index.isDefined
+      })
+  }
+
+  private def hasLogs(entries: Iterable[ExecutionDatabaseKey])(key: ExecutionDatabaseKey) = {
+    !key.fqn.isScatter && !isCollector(entries)(key)
```

Comment:
Yes I didn't really want to do that but I thought changing FullyQualifiedName from `type String` to `case class` was a pretty big deal.
On the other hand I wanted to avoid referencing the "$scatter" string in the WorkflowActor which would I think be leaking even more binding to engine.
This is why I ended up with this.